extension/js/common/core/openpgp-key.ts

```diff
@@ -79,12 +136,12 @@ export class OpenPGPKey {
       emails,
       // tslint:disable-next-line: no-unsafe-any
       identities: pubkey.users.map(u => u.userId).filter(u => !!u && u.userid && Str.parseEmail(u.userid).email).map(u => u!.userid).filter(Boolean) as string[],
-      lastModified: new Date(await PgpKey.lastSigOpenPGP(pubkey)),
+      lastModified,
       expiration: exp instanceof Date ? exp : undefined,
       created: pubkey.getCreationTime(),
       checkPassword: _text => Promise.resolve(false),
-      fullyDecrypted: pubkey.isFullyDecrypted(),
-      fullyEncrypted: pubkey.isFullyEncrypted(),
+      fullyDecrypted: pubkey.isPublic() ? true /* public keys are always decrypted */ : pubkey.isFullyDecrypted(),
```

Comment:
I think, if anything, public keys should show "undefined", as in not applicable

Reply:
Yep, sounds good. I saw that these two only make sense for private keys but if I'm not mistaken some code checks for these properties on public keys too (maybe this is a side effect). I'll diagnose it further when I get the tests in shape.

Also: there are some optimizations when a key is `encrypt`ed or `decrypt`ed as I think only these two properties change.... this is WIP still :)